core/shared/src/main/scala/zio/prelude/Traversable.scala

```diff
@@ -122,6 +122,30 @@ trait Traversable[F[+_]] extends Covariant[F] {
       }
     }
 
+  /**
+   * Compute each element of the collection
+   * with specified effectual function `f`
+   * and then reduces the results using the
+   * `AssociativeEither[G].both` operation,
+   * returning `None` if the collection is empty.
+   */
+  def forany[G[+_]: AssociativeEither: Covariant, A, B](fa: F[A])(f: A => G[B]): Option[G[B]] = {
```

Comment:
Isn't this `collectFirst`? It applies the function` f` to each element in the structure and returns the first result that is a "success", where the meaning of success and failure is defined by the `AssociativeEither` instance?

Reply:
Yes, that's what I was going for! Thank you @adamgfraser , I didn't know it could have this name :+1: 
https://superruzafa.github.io/visual-scala-reference/collectFirst/
Although I would suspect we might need `IdentityEither` for this.

I was pushed to make a proposal for it, when I was looking at the signature for `foreach` and thought there should be a dual for it -- hence the name `forany`.

Reply:
@sideeffffect Wow! That Visual Scala resource is really cool!

I think we can get away with just having an `AssociativeEither` instance because `None` can serve as the default value if the structure is empty or if there is no successful value. However, I think we need to flip the order of the result types here so we return a `G[Option[B]]`. Otherwise we end up returning the last "failure" if the structure is non-empty and all of the results are failures.

Reply:
Yes. Good point! But in that case, we would probably need `IdentityBoth`, to be able to create a `G` of `None` in case of an empty Traversable, right?
I've updated the ticket description https://github.com/zio/zio-prelude/issues/392
Please let me know, if you consider it accurate now :pray: